cme/protocols/smb/wmiexec.py

```diff
@@ -8,121 +8,122 @@
-                    self.output_callback(output.read())
-                break
-            except IOError:
-                sleep(2)
-
-    def get_output_remote(self):
-        if self.__retOutput is False:
-            self.__outputBuffer = ''
-            return
-
-        while True:
-            try:
-                self.__smbconnection.getFile(self.__share, self.__output, self.output_callback)
-                break
-            except Exception as e:
-                if str(e).find('STATUS_SHARING_VIOLATION') >=0:
-                    # Output not finished, let's wait
-                    sleep(2)
-                    pass
-                else:
-                    #print str(e)
-                    pass
-
-        self.__smbconnection.deleteFile(self.__share, self.__output)
+	def __init__(self, target, share_name, username, password, domain, smbconnection, hashes=None, share=None):
+		self.__target = target
+		self.__username = username
+		self.__password = password
+		self.__domain = domain
+		self.__lmhash = ''
+		self.__nthash = ''
+		self.__share = share
+		self.__smbconnection = smbconnection
+		self.__output = None
+		self.__outputBuffer = ''
+		self.__share_name = share_name
+		self.__shell = 'cmd.exe /Q /c '
+		self.__pwd = 'C:\\'
+		self.__aesKey = None
+		self.__doKerberos = False
+		self.__retOutput = True
+
+		if hashes is not None:
+		#This checks to see if we didn't provide the LM Hash
+			if hashes.find(':') != -1:
+				self.__lmhash, self.__nthash = hashes.split(':')
+			else:
+				self.__nthash = hashes
+
+		if self.__password is None:
+			self.__password = ''
+
+		self.__dcom = DCOMConnection(self.__target, self.__username, self.__password, self.__domain, self.__lmhash, self.__nthash, self.__aesKey, oxidResolver = True, doKerberos=self.__doKerberos)
+		iInterface = self.__dcom.CoCreateInstanceEx(wmi.CLSID_WbemLevel1Login,wmi.IID_IWbemLevel1Login)
+		iWbemLevel1Login = wmi.IWbemLevel1Login(iInterface)
+		iWbemServices= iWbemLevel1Login.NTLMLogin('//./root/cimv2', NULL, NULL)
+		iWbemLevel1Login.RemRelease()
+
+		self.__win32Process,_ = iWbemServices.GetObject('Win32_Process')
+
+	def execute(self, command, output=False):
+		self.__retOutput = output
+		if self.__retOutput:
+			self.__smbconnection.setTimeout(100000)
+
+		self.execute_handler(command)
+		self.__dcom.disconnect()
+		return self.__outputBuffer
+
+	def cd(self, s):
+		self.execute_remote('cd ' + s)
+		if len(self.__outputBuffer.strip('\r\n')) > 0:
+			print(self.__outputBuffer)
+			self.__outputBuffer = ''
+		else:
+			self.__pwd = ntpath.normpath(ntpath.join(self.__pwd, s))
+			self.execute_remote('cd ')
+			self.__pwd = self.__outputBuffer.strip('\r\n')
+			self.__outputBuffer = ''
+
+	def output_callback(self, data):
+		self.__outputBuffer += data.decode("utf-8") 
+
+	def execute_handler(self, data):
+		if self.__retOutput:
+			try:
+				logging.debug('Executing remote')
+				self.execute_remote(data)
```

Comment:
for the purpose of the test, I use `execute_remote` to get result of my command on Win10.